data/transactions/logic/eval.go

```diff
@@ -251,10 +251,21 @@ type LedgerForLogic interface {
 	Counter() uint64
 }
 
-// boxRef is the "hydrated" form of a BoxRef - it has the actual app id, not an index
-type boxRef struct {
-	app  basics.AppIndex
-	name string
+// BoxRef is the "hydrated" form of a transactions.BoxRef - it has the actual app id, not an index
+type BoxRef struct {
+	App  basics.AppIndex
+	Name string
+}
+
+// UnnamedResourcePolicy is an interface that defines the policy for allowing unnamed resources.
+// This should only be used during simulation or debugging.
+type UnnamedResourcePolicy interface {
```

Comment:
There seems to be only one implementor. Is it an interface in order to avoid circular dependencies?

Reply:
Correct. That and I felt the implementing code really ought to live in the simulation package, since its behavior is inherently tied to what the simulator wants.